stake-pool/js/src/index.ts

```diff
@@ -649,6 +649,58 @@ export async function withdrawSol(
   };
 }
 
+export async function addValidatorToPool(
+  connection: Connection,
+  stakePoolAddress: PublicKey,
+  validatorVote: PublicKey,
+  seed?: number,
+) {
+  const stakePool = await getStakePoolAccount(connection, stakePoolAddress);
+
+  const validatorList = await getValidatorListAccount(
+    connection,
+    stakePool.account.data.validatorList,
+  );
+
+  const validatorInfo = validatorList.account.data.validators.find(
+    (v) => v.voteAccountAddress.toBase58() == validatorVote.toBase58(),
+  );
+
+  if (validatorInfo) {
+    throw new Error('Vote account is already in validator list');
+  }
+
+  const withdrawAuthority = await findWithdrawAuthorityProgramAddress(
+    STAKE_POOL_PROGRAM_ID,
+    stakePoolAddress,
+  );
+
+  const validatorStake = await findStakeProgramAddress(
+    STAKE_POOL_PROGRAM_ID,
+    validatorVote,
+    stakePoolAddress,
+    seed ? new BN(seed) : undefined,
+  );
+
+  const instructions: TransactionInstruction[] = [];
+
+  instructions.push(
+    StakePoolInstruction.addValidatorToPool({
+      stakePool: stakePoolAddress,
+      staker: stakePool.account.data.staker,
+      reserveStake: stakePool.account.data.reserveStake,
+      withdrawAuthority: withdrawAuthority,
+      validatorList: stakePool.account.data.validatorList,
+      validatorStake: validatorStake,
+      validatorVote: validatorVote,
+    }),
+  );
+
+  return {
+    instructions,
+  };
```

Comment:
Since we don't need to dynamically push anything else into this list, why not just return the single-element list as-is, without doing `instructions.push()`?

Reply:
Agreed and done